src/components/SeedInfo/SeedInfoViews/HolyMountain.tsx

```diff
@@ -135,16 +136,36 @@ const PacifistChest: FC<IPacifistChestProps> = ({ items }) => {
 // TODO: Extract this into it's own file to decouple
 const Shop = ({ type, handleOpenShopInfo, favoriteSpells }) => {
   const Icon = type === IShopType.wand ? WandIcon : LightBulletIcon;
+
+  // group duplicate favorite spells and count occurrences
+  const favoriteSpellsGrouped : Map<string, number> = favoriteSpells.reduce((acc, spell) => {
+    acc.set(spell, (acc.get(spell) ?? 0) + 1); return acc;
+  }, new Map<string, number>());
+  
+
+  // add favorite spell icons with badges for counts
+  const favSpellIcons = Array.from(favoriteSpellsGrouped.entries()).map(([spell, count]) => (
+    <BadgesWrapper key={spell} badges={[CountBadge({ text: count.toString() })]}>
+      <Entity
+        key={spell}
+        width="1rem"
+        height="1rem"
+        id="Spell"
+        entityParams={{ extra: spell }}
+      />
+    </BadgesWrapper>
+  ));
+
   return (
     <Button
       className="position-relative"
       onClick={handleOpenShopInfo}
       variant={favoriteSpells.length ? "outline-info" : "outline-primary"}
       size="sm"
     >
+      {!favSpellIcons.length ? "" : <div className="d-flex">{favSpellIcons}</div>}
```

Comment:
nit: this can be simplified to favSpellIcons.length && <div className="d-flex">{favSpellIcons}</div>